### ISSUE TITLE
When should we change `main` to use 0.9.0 version number?

### ISSUE THREAD

**Issue by richb-hanover** [PRQL/prql#2957]
### What's up?

When we release 0.9.0, obviously the version number reported in the generated SQL will switch to the new version.

When is the proper time to do this in the run-up to that time? Thanks.

**Comment by aljazerzen** [comment]
This is a bit problematic when posting queries from a development build, which is (as it is now) close to 0.9 release, but it is still reporting version 0.8.1.

But to be precise, it is not the 0.9 version yet.

Some projects report `0.8.1-15` where 15 is number of commits since 0.8.1.

Some projects report `0.8.1-360dacab` where is the git hash of the commit.

Some projects report `0.8.1-dev`.

I'm fine with any of these, but I don't want to invest the time to set it up :D

**Comment by richb-hanover** [comment]
> I'm fine with any of these, but I don't want to invest the time to set it up :D

And I am fine with not adding any more work for you :-) I just asked in case someone wanted to say, "Yes, let's do it now!"

**Comment by aljazerzen** [comment]
Let's leave this open if anyone is up for the task

**Comment by max-sixty** [comment]
I looked into this (seems like I didn't post an issue though).

I think we could bump the version to — for example — `0.8.2-pre` or `0.8.2-dev` after releasing `0.8.1`. (Or `cargo-release` [handles](https://github.com/crate-ci/cargo-release/blob/master/docs/reference.md) `-alpha` / `-beta` slightly more natively, but that would maybe imply something different).

So that would require:
1. Another instruction in the release notes, to run:
    ```
    cargo release version ${NEXT_VERSION}-pre -x --no-confirm && cargo release replace -x --no-confirm
    ```

    ...where `$NEXT_VERSION` is typed in.

2. We'd need to update the release replacements (e.g. at https://github.com/PRQL/prql/blob/cdce7ad390949d99fdda9874650bc9fe90893a13/bindings/prql-js/Cargo.toml#L50-L56, and in some other Cargo.toml files) to allow non-digit characters. 
  
    If someone wants to take a swing at that, that would be welcome. 
    To test, run this _after_ running the command above with `-pre` above and confirm it does replacements correctly.

    ```
    cargo release version patch -x --no-confirm && cargo release replace -x --no-confirm
    ```

**Comment by eitsupi** [comment]
Closed by #3180
